### PR TITLE
Integrate `BaseApiClient` into the `Client`

### PR DESCRIPTION
**Pull request by ghost**
This PR adresses #26. 

_Note: The tests are still failing and some more modifications need to be made to the code, thus the draft PR._

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -31,7 +31,7 @@ dependencies = [
   "googleapis-common-protos >= 1.56.4, < 2",
   "grpcio >= 1.60.0, < 2",
   "frequenz-channels >= 1.0.0, < 2",
-  "frequenz-client-base >= 0.3.0, < 0.4",
+  "frequenz-client-base >=0.5",
```
Comment (Marenz) [inline pyproject.toml]: I think the dependency should now include the grpc dependency, like so

```suggestion
  "frequenz-client-base[grpcio] >= 0.5.0, < 0.6.0",  
```

File: src/frequenz/client/electricity_trading/_client.py [hunk 2]
```diff
@@ -346,7 +347,7 @@ async def create_gridpool_order(  # pylint: disable=too-many-arguments, too-many
         try:
             response = await cast(
                 Awaitable[electricity_trading_pb2.CreateGridpoolOrderResponse],
-                self._stub.CreateGridpoolOrder(
+                self.stub.CreateGridpoolOrder(
                     electricity_trading_pb2.CreateGridpoolOrderRequest(
                         gridpool_id=gridpool_id,
                         order=order.to_pb(),
```
Comment (Marenz) [inline src/frequenz/client/electricity_trading/_client.py:354]: As far as I understood the documentation and example, the intended way to call functions now is
```
response = await call_stub_method(self, async lambda: await self.stub.CreateGridpoolOrder(..))
```

Comment (Marenz) [reply src/frequenz/client/electricity_trading/_client.py:354]: Nevermind that part though, apparently `call_stub_method` isn't easily usable with async channels

File: src/frequenz/client/electricity_trading/_client.py [hunk 3]
```diff
@@ -112,6 +117,7 @@ def __init__(self, grpc_channel: grpc.aio.Channel) -> None:
                 electricity_trading_pb2.ReceivePublicTradesStreamResponse, PublicTrade
             ],
         ] = {}
+        self._metadata = (("key", key),) if key else ()
```
Comment (matthias-wende-frequenz) [inline src/frequenz/client/electricity_trading/_client.py]: This is not sufficient to cover key based authorization. I suggest to remove this line and add all the necessary things in a follow up PR.

File: src/frequenz/client/electricity_trading/_client.py (hunk 2 above)
Comment (matthias-wende-frequenz) [reply src/frequenz/client/electricity_trading/_client.py:354]: for all the non streaming methods that seems to be the way to go though.

Comment (matthias-wende-frequenz) [reply src/frequenz/client/electricity_trading/_client.py:354]: and for streaming there is already the `GrpcStreamBroadcaster` in place.

**Review by matthias-wende-frequenz — COMMENTED** [review]
I've added a few comments.

File: src/frequenz/client/electricity_trading/_client.py (hunk 2 above)
Comment (ghost) [reply src/frequenz/client/electricity_trading/_client.py:354]: Not sure I get this right, but if it doesn't work yet I'd rather wait before we implement it here

Comment (llucax) [reply src/frequenz/client/electricity_trading/_client.py:354]: Yeah, if it doesn't work it is a bug, and probably due to this never being used with grpcio before. I'm currently busy with some other higher prio stuff, so if nobody else can have a look at it now we'll have to delay this PR a bit until I have some time to see what's wrong/missing with grpcio for using `call_stub_method`).

Comment (llucax) [reply src/frequenz/client/electricity_trading/_client.py:354]: **OR** we can actually not use `call_stub_method` yet, having the client integrated without it still has value, as at least we would get a unified way to constructur/connect/close clients and use them as context managers.

File: tests/test_client.py [hunk 4]
```diff
@@ -39,8 +39,7 @@
 def set_up() -> Generator[Any, Any, Any]:
     """Set up the test suite."""
     # Create a mock client and stub
-    mock_channel = MagicMock()
-    client = Client(grpc_channel=mock_channel)
+    client = Client("grpc://unknown.host", connect=False)  # noqa: F841
```
Comment (llucax) [inline tests/test_client.py]:
```suggestion
    _ = Client("grpc://unknown.host", connect=False)
```

File: src/frequenz/client/electricity_trading/_client.py (hunk 2 above)
Comment (matthias-wende-frequenz) [reply src/frequenz/client/electricity_trading/_client.py:354]: Then I'd suggest to move forward without this!

**Review by matthias-wende-frequenz — APPROVED** [review]

**Pull request merged**